lib/mpool/include/hse/mpool/mpool.h

```diff
@@ -457,9 +457,45 @@ mpool_mblock_read(struct mpool *mp, uint64_t mbid, const struct iovec *iov, int
 merr_t
 mpool_mblock_clone(struct mpool *mp, uint64_t mbid, off_t off, size_t len, uint64_t *mbid_out);
 
+/**
+ * mpool_mblock_punch() - Punch a hole into the specified region of a committed mblock
+ *
+ * @mp:   mpool
+ * @mbid: mblock object ID to punch
+ * @off:  start offset
+ * @len:  number of bytes to punch
+ *
+ * Requirements on `off' and `len':
+ * - off must not be less than 0
+ * - off and len must be page aligned
+ * - len == 0 implies a punch of the entire mblock from 'off' upto its written length
+ * - off + len must not be greater than wlen
+ *
+ * Notes:
+ * - Reading the mblock from the punched region returns zeroes
+ * - wlen is updated if the punched region covers mblock's written length
```

Comment:
I would write "Reading *an* mblock from the punched region returns zeroes"

I would also write "The mblock's wlen property is updated" (which I assume to be the case since there is no output named wlen).

I presume wlen is updated when the punched region covers the tail end of the written len?

Is the allocated len property updated?  I guess not.

The reason for calling this is to free up memory and logically truncate the mblock as opposed to freeing up media storage?  If so, might be worth adding that to the function comment (along with the fact that the alen is not changed).

Reply:
Will incorporate your feedback to the comments.

Yes, wlen is updated when the punched region covers the tail end.
alen is also updated to account for the punched region. So the media storage is freed up when calling this function and that's what will enable kvset splits to not generate vblock garbage.

Example 1:
Before punch: alen= 32M, wlen=32M
punch(mblock, 0, 16M)
After punch: alen = 16M, wlen = 32M

Example 2:
Before punch: alen= 32M, wlen=32M
punch(mblock, 16M, 0)
After punch: alen = 16M, wlen = 16M

Example 3:
Before punch: alen= 32M, wlen=32M
punch(mblock, 16M, 8M)
After punch: alen = 24M, wlen = 32M